src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,371 @@
+# (Hairer & Wanner 1996, Vol I, p.398-400, 421-423)
+# ODE_MS Fixed-step, fixed-order multi-step numerical method
+#   with Adams-Bashforth-Moulton coefficients
+function ode_ab_adaptive(F, y0, tspan, order::Integer;order_selection = false,
+                                                      reltol = 1.0e-5,
+                                                      abstol = 1.0e-8,
+                                                      minstep=abs(tspan[1] - tspan[end])/1e12,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.4,
+                                                      facmin = .1,
+                                                      fac = .8)
+
+    # initialization of time span, which will have variable length
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    y = typeof(y0)[]
+    push!(y,y0)
+
+    # initialzation of dy[n], which will have variable length as well
+    dy = typeof(F(t[1], y[1]))[]
+    push!(y,F(t[1], y[1]))
+
+    #initialization of step size array, which will have variable length as well
+    h = [initstep]
+    if h[1] == 0.
+        # initial guess at a step size
+        h[1], tdir, dy0 = hinit(F, y0, tstart, tfinal, 3, reltol, abstol)
+    else
+        tdir = sign(tfinal - tstart)
+    end
+
+    #uncomment line below to test with fixed step size
+    #@show h = diff(tspan)
+
+    #initialization of arrays used in calculation of further steps
+    #all initialized to a order x order array, which will need to expand to
+    # n x order over course of algorithm, as we use n steps, where n is
+    #unknown
+    Tdy=typeof(F(t[1], y[1]))
+    c,g,b,Φstar, Φ = intialize_coef_arrays(order+3,Tdy)
+
+    #Array which holds norms of LE(n+1), the local error or truncation error, which will
+    #will used in step size selection. normLE[1] corresponds to norm of LE(n+1)_{k-1},
+    # normLE[2] to norm of LE(n+1)_{k},  normLE[3] to norm of  LE(n+1)_{k+1},
+    #and normLE[4] to norm of  LE(n+1)_{k+2},
+    normLE = [0.0 0.0 0.0 0.0]
+
+    ##Note that due to Julia starting indexing at 1, k = steporder + 1 in what follows
+    #start at n=1, the first step
+    n = 1
+    steporder=1
+    wait=0
+    while abs(t[n] - tfinal) > 0 && sign(tfinal-t[n])==tdir #&&  minstep < abs(h[n])
+        # Need to run the first several steps at reduced order, and need to maintain
+        # n>steporder invariant in order to perform adaptive step selection of the
+        # 'order' = steporder, for it requires the last 'order' number of points
+        if n<order
+            steporder = n==1 ? 1 : min(n-1, order)
+        end
+
+        #allocate space for t[n+1], y[n+1], dy[n+1], h[n+1]
+        t,y,dy, h = increment_length(t,y,dy,h,n)
```

Comment:
Any function which modifies its arguments inplace should have a `!` at its end.  Also, you don't need to return.
```
increment_length!(t,y,dy,h,n)

....
function increment_length!(t,y,dy,h,n)
    resize!(t,n+1)
    resize!(y,n+1)
    resize!(dy,n+1)
    resize!(h,n+1)
    return nothing
end
```